internal/controller/configmap.go

```diff
@@ -86,12 +85,16 @@ func buildPolicyStringRules(role *argoprojiov1alpha1.Role, roleName string) stri
 }
 
 // buildPolicyStringSubjects will build the policy string for Subjects field of the given role.
-func buildPolicyStringSubjects(role *argoprojiov1alpha1.Role, roleName string) string {
+func buildPolicyStringSubjects(rb *argoprojiov1alpha1.ArgoCDRoleBinding, role *argoprojiov1alpha1.ArgoCDRole) string {
 	policy := ""
-	for _, subject := range role.Spec.Subjects {
+	roleName := fmt.Sprintf("role:%s", role.ObjectMeta.Name)
+	for _, subject := range rb.Spec.Subjects {
 		switch subject.Kind {
 		case "sso":
 			policy += fmt.Sprintf("g, %s, %s\n", subject.Name, roleName)
+		case "role":
+			subjectRoleName := fmt.Sprintf("role:%s", subject.Name)
+			policy += fmt.Sprintf("g, %s, %s\n", subjectRoleName, roleName)
 		case "local":
 			policy += buildPolicyStringRules(role, subject.Name)
```

Comment:
I think this one should be handled by the role itself. It basically duplicates the role under a new name, I don't know how I feel about it

Reply:
I think this approach is better since u only have to change one resource to give/revoke permissions, if we setup localUser in Role, u'd have to change both

Reply:
all good! for my usecase, I'll only need sso and role